### ISSUE TITLE
[ENHANCEMENT] Getting started docs are hard to follow

### ISSUE THREAD

**Issue by r0fls** [open-telemetry/opentelemetry.io#1748]
**What needs to be changed?**
The getting started docs should require minimal reading and comprehension. The estimated reading time is 5-10 minutes for this document https://opentelemetry.io/docs/instrumentation/go/getting-started/

Likely longer, if you care to look up each term here that you're unfamiliar with (imagine reading this for the first time, never having heard of tracing). These would be better suited for a page called something like "introductory concepts" or something along those lines.

IMO: it should take ~1 minute. This is not a good place to teach concepts like "Installing a Tracer Provider", "Creating a Resource", "Creating a Console Exporter", etc... it should be a code snippet that you can yank, fill in a tracing provider URL, and it will start exporting traces.

**What is the name + path of the page that needs changed?**
The relative path and page title where you found a problem.
https://opentelemetry.io/docs/instrumentation/go/getting-started/

Possibly the getting started guides for the other languages.

**Additional context**
This is just my opinion, but these docs could seriously be enhanced. Having to copy the code from each snippet and try to figure out how all of the snippets are meant to be assembled is not a good way to write getting started guides: we should have something closer to a single working snippet that a user can copy.... copying separate snippets requires a trust that it will work once we assemble all of these components together, otherwise I will be wasting my time. If it doesn't work, we essentially break that trust right here in our getting started guide.

**Comment by cartermp** [comment]
Here is where I make my shame public for not following up on some of the latest comments here: https://github.com/open-telemetry/opentelemetry-go/pull/2895

It gets things to be a lot closer to what you're talking about here.

That said, configuring opentelemetry "correctly" does involve quite a bit of code, and so while it is copy-pasteable, it is not particularly easy. There is a need to explain some of the concepts because a lot of new programmers won't just take 30+ lines of configuration code at face value. But it's a lot more succinctly described, at least.

**Comment by svrnm** [comment]
+1 to what @cartermp said. 

To add a generic comment: I think we all agree that the opentelemetry documentation requires some love to be more userfriendly now that lots of things are stable and many SIGs are also looking into means to make things easier (especially through auto instrumentation). So, to make a long story short: we eagerly require people to help us to make those docs better step by step. So, take a page, and provide some incremental improvements, @cartermp and I are here to get your work integrated quickly

**Comment by svrnm** [comment]
Most/Some of the concerns are addressed by #2623, @r0fls please take a look what has been changed since you raised this issue and if you think more needs to be done and comment on that meta issue. Thank you!
